src/test/java/uk/gov/pay/api/utils/mocks/ChargeResponseFromConnector.java

```diff
@@ -0,0 +1,198 @@
+package uk.gov.pay.api.utils.mocks;
+
+import uk.gov.pay.api.model.CardDetails;
+import uk.gov.pay.api.model.PaymentState;
+import uk.gov.pay.api.model.RefundSummary;
+import uk.gov.pay.api.model.SettlementSummary;
+import uk.gov.pay.commons.model.SupportedLanguage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class ChargeResponseFromConnector {
+    public final Long amount, corporateCardSurcharge, totalAmount; 
+    public final PaymentState state; 
+    public final String chargeId, returnUrl, description, reference, email, paymentProvider, gatewayTransactionId, createdDate;
+    public final SupportedLanguage language; 
+    public final boolean delayedCapture; 
+    public final RefundSummary refundSummary; 
+    public final SettlementSummary settlementSummary; 
+    public final CardDetails cardDetails;
+    public final List<Map<?, ?>> links;
+    public final Optional<Map<String, Object>> metadata;
+
+    private ChargeResponseFromConnector(long amount, String chargeId, PaymentState state, String returnUrl,
```

Comment:
Sure there's benefits either way but since this constructor is private and can only be created via a builder then it could just take the instance of the builder as the argument rather than having to deconstruct it and have lots of arguments? e.g. `private ChargeResponseFromConnector(ChargeResponseFromConnectorBuilder builder)`. This is just my opinion and I would accept it as-is if you prefer.